azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -383,39 +391,94 @@ private OSProfile createOsProfile(String computerName, Template template) {
+         }
 
-      final NetworkInterfaceCardProperties.Builder networkInterfaceCardProperties = NetworkInterfaceCardProperties
-              .builder()
-              .ipConfigurations(
-                      of(IpConfiguration
-                              .builder()
-                              .name("ipConfig-" + name)
-                              .properties(
-                                      IpConfigurationProperties.builder().privateIPAllocationMethod("Dynamic")
-                                              .publicIPAddress(IdReference.create(ip.id())).subnet(IdReference.create(subnetId))
-                                              .build()).build()));
+         String ipName = namingConvention.uniqueNameForGroup(subnetName);
+         final String nicName = namingConvention.uniqueNameForGroup(subnetName);
 
-      String securityGroup = getOnlyElement(options.getGroups(), null);
-      if (securityGroup != null) {
-         networkInterfaceCardProperties.networkSecurityGroup(IdReference.create(securityGroup));
+         IpConfiguration config = IpConfiguration.builder()
+               .name(ipName)
+               .properties(ipProperties.build())
+               .build();
+
+         NetworkInterfaceCardProperties.Builder nicProperties = NetworkInterfaceCardProperties.builder()
+               .ipConfigurations(ImmutableList.of(config));
+         
+         if (securityGroup != null) {
+            nicProperties.networkSecurityGroup(IdReference.create(securityGroup));
+         }
+
+         logger.debug(">> creating nic %s(%s) with security groups (%s)", nicName, config,
+               securityGroup != null ? securityGroup : "");
+         
+         final NetworkInterfaceCardApi nicApi = api.getNetworkInterfaceCardApi(resourceGroup);
+         NetworkInterfaceCard nic = nicApi.createOrUpdate(nicName, location, nicProperties.build(),
+               ImmutableMap.of("jclouds", nodeName));
+         
+         resourceAvailable.apply(new Supplier<Provisionable>() {
+            @Override
+            public Provisionable get() {
+               NetworkInterfaceCard updated = nicApi.get(nicName);
+               return updated == null ? null : updated.properties();
+            }
+         });
+
+         nics.add(nic);
       }
 
-      String networkInterfaceCardName = "jc-nic-" + name;
-      return api.getNetworkInterfaceCardApi(azureGroup).createOrUpdate(networkInterfaceCardName, locationName,
-              networkInterfaceCardProperties.build(), ImmutableMap.of("jclouds", name));
+      return nics.build();
+   }
+   
+   private NetworkProfile createNetworkProfile(List<NetworkInterfaceCard> nics) {
+      List<NetworkInterface> nicAttachments = new ArrayList<NetworkInterface>(nics.size());
+      for (int i = 0; i < nics.size(); i++) {
+         nicAttachments.add(NetworkInterface.create(nics.get(i).id(), NetworkInterfaceProperties.create(i == 0)));
```

Comment:
maybe a javadoc to explain the method as it is not obvious, especially the `i == 0` part